### PR TITLE
Add Export functionality

### LINKED ISSUE #74: Missing Export
Export feature is missing

### PR DESCRIPTION
**Pull request by devinchristianson**
My apologies this ended up being a larger PR, turns out there was more to getting this going than we initially appreciated. A quick list of what was changed:

- Added PrivateStorageDetailView to serve the files
- Proxied /rawdata/ through the nginx in order to serve via send_file with the nginx
- Added export view that figures out whether to export a tiff or zip for the given dataset
- Added file name tracking to the rawdata model so that downloads have reasonable names
- Updated permissions to that all the services use compatible permissions
- Removed `makemigrations` from the initial setup script because a) it was incompatible with the new permissions structure and b) it never really should have been there (it's dangerous to makemigrations on a prod host)

### CODE REVIEW

**Comment by NickKania** [comment]
I am getting Permission denied when attempting to upload a dataset [
Screen Shot 2021-04-24 at 10 06 44 PM
](url)

File: birdspotter/birdspotter/dataio/views.py [hunk 1]
```diff
@@ -88,3 +89,46 @@ def share_dataset(request, dataset_id):
...
+@login_required
+def export_dataset(request, dataset_id):
+    """
+    Handles the export dialog and accepts GET and POST requests to /export/<dataset_id>/
+    POST: Get list of potential raw_data's
+    GET: Returns an object with an array of usernames, along with if they are already shared the dataset
+    """
+    dataset = Dataset.objects.get(dataset_id=dataset_id)
+    # only allow owner of dataset to share the dataset
+    if dataset.owner == request.user or request.user.is_admin() or request.user.groups.filter(name=GROUPS.privileged).exists():
```
[the remark this answers is not in the document]
Comment (devinchristianson) [reply birdspotter/birdspotter/dataio/views.py:103]: This is actually a good question - based on the SRS etc, all logged in users are able to download all datasets... idk if this should be the case, but if we restrict downloads to privileged users, or even further, we'd need to add checking dataset sharing

File: docker-compose/docker-compose.override.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ services:
     volumes:
       - "/var/run/docker.sock:/var/run/docker.sock"
     command:
+      - "--log.level=DEBUG"
```
[the remark this answers is not in the document]
Comment (devinchristianson) [reply docker-compose/docker-compose.override.yml]: Whoops, removed

**Comment by devinchristianson** [comment]
> 
> 
> I am getting Permission denied when attempting to upload a dataset [ Screen Shot 2021-04-24 at 10 06 44 PM ](url)

Have you re-built your docker images before trying to run this? If not, let me know if it works post-rebuild, since I didn't run into this. If you have re-built it, are you running in the default dev mode, or in prod mode (by specifying the prod docker files)?

**Review by NickKania — APPROVED** [review]

**Pull request merged**